File: core/python/models/filing.py
"""Pydantic models for SEC filings."""

from datetime import datetime

from pydantic import BaseModel


class FilingCreate(BaseModel):
    """Data required to insert a new filing."""

    accession_number: str
    form_type: str
    filed_at: datetime
    accepted_at: datetime | None = None
    filer_cik: str
    filer_name: str
    url: str
    metadata: dict = {}


class Filing(FilingCreate):
    """A filing as stored in the database."""

    id: int
    full_text: str | None = None
    embedding: list[float] | None = None
    processed: bool = False
    created_at: datetime

File: frontend/src/schemas/filing.ts
import { z } from "zod";

export const filingCreateSchema = z.object({
  accession_number: z.string().min(1),
  form_type: z.string().min(1),
  filed_at: z.string(),
  accepted_at: z.string().nullable().default(null),
  filer_cik: z.string().min(1),
  filer_name: z.string().min(1),
  url: z.string().min(1),
  metadata: z.record(z.string(), z.unknown()).default({}),
});

export const filingSchema = filingCreateSchema.extend({
  id: z.number().int(),
  full_text: z.string().nullable().default(null),
  embedding: z.array(z.number()).nullable().default(null),
  processed: z.boolean().default(false),
  created_at: z.string(),
});

export type FilingCreate = z.infer<typeof filingCreateSchema>;
export type Filing = z.infer<typeof filingSchema>;
